packages/store/src/StoreView.sol

```diff
@@ -2,74 +2,42 @@
 pragma solidity >=0.8.0;
 
 import { SchemaType } from "./Types.sol";
-import { IStore, IStoreHook } from "./IStore.sol";
+import { IStoreHook } from "./IStore.sol";
 import { StoreCore } from "./StoreCore.sol";
 import { Schema } from "./Schema.sol";
+import { Store } from "./Store.sol";
 
 // Not abstract, so that it can be used as a base contract for testing and wherever write access is not needed
-contract StoreView is IStore {
+contract StoreView is Store {
   error StoreView_NotImplemented();
 
-  constructor() {
-    StoreCore.initialize();
-  }
-
-  function getSchema(uint256 table) public view virtual returns (Schema schema) {
-    schema = StoreCore.getSchema(table);
-  }
-
   function registerSchema(uint256, Schema) public virtual {
     revert StoreView_NotImplemented();
   }
 
   function setRecord(
     uint256,
-    bytes32[] memory,
-    bytes memory
+    bytes32[] calldata,
+    bytes calldata
   ) public virtual {
     revert StoreView_NotImplemented();
   }
 
   // Set partial data at schema index
   function setField(
     uint256,
-    bytes32[] memory,
+    bytes32[] calldata,
     uint8,
-    bytes memory
+    bytes calldata
   ) public virtual {
     revert StoreView_NotImplemented();
   }
 
-  function registerHook(uint256, IStoreHook) public virtual {
+  function registerStoreHook(uint256, IStoreHook) public virtual {
```

Comment:
since not just arguments, but this whole method isn't usable, I rather think it's better for docs to add something like
```solidity
/**
 * Not implemented in StoreView
 */
```